src/main_nep/nep3.cu

```diff
@@ -187,7 +187,11 @@ static __global__ void find_descriptors_angular(
         }
         accumulate_s(d12, x12, y12, z12, gn12, s);
       }
-      find_q_nep3(paramb.n_max_angular + 1, n, s, q);
+      if (paramb.num_L > paramb.L_max) {
+        find_q_with_4body(paramb.n_max_angular + 1, n, s, q);
+      } else {
+        find_q(paramb.n_max_angular + 1, n, s, q);
+      }
```

Comment:
What is the difference between `src/main_nep/nep3.cu` and `src/force/nep3.cu`? Does the former only predict energy, and the latter only forces?

Reply:
src/main_nep/nep3.cu is used for the `nep` executable, that is, the NEP training code.
src/force/nep3.cu is used for the `gpumd` executable, that is, the MD production code.

They need to be different because they data structures are quite different (only a single box in MD but many structures in training).